### ISSUE TITLE
CID 47534: Pointer to local outside scope (CWE-562)

### ISSUE THREAD

**Issue by mmaraya** [mmaraya/netkit-ftp#18]
[ftp/cmds.c#L701](https://github.com/mmaraya/netkit-ftp/blob/7c5f7827dac249adf3aad87047915bbedc1c46e6/ftp/cmds.c#L701)
[ftp/cmds.c#L703](https://github.com/mmaraya/netkit-ftp/blob/7c5f7827dac249adf3aad87047915bbedc1c46e6/ftp/cmds.c#L703)
[ftp/cmds.c#L708](https://github.com/mmaraya/netkit-ftp/blob/7c5f7827dac249adf3aad87047915bbedc1c46e6/ftp/cmds.c#L708)
[ftp/cmds.c#L768](https://github.com/mmaraya/netkit-ftp/blob/7c5f7827dac249adf3aad87047915bbedc1c46e6/ftp/cmds.c#L768)

```
use_invalid: Using argv[2], which points to an out-of-scope variable tmpbuf.
```

Dereferencing the returned or out-of-scope stack pointer will access an invalid location on the stack after its scope or after the function returns. In `get`: Pointer to a local stack variable returned or used outside scope [(CWE-562)](https://cwe.mitre.org/data/definitions/562.html)

**Comment by mmaraya** [comment]
Coverity reports rest are fixed but [ftp/cmds.c#L703](https://github.com/mmaraya/netkit-ftp/blob/7c5f7827dac249adf3aad87047915bbedc1c46e6/ftp/cmds.c#L703) remains open:

```
escape_local_addr_alias: Returning, through argv[2], pointer domap(argv[2]) which points to local variable tmpbuf.
```

**Comment by mmaraya** [comment]
Remaining bug is a false positive; all four instances of this bug have been fixed by c1ca7ce
